### ISSUE TITLE
🐛 [cloud_functions] The plugin `cloud_functions` uses a deprecated version of the Android embedding.

### ISSUE THREAD

**Issue by dark-chocolate** [FirebaseExtended/flutterfire#6770]
## Bug report

Run app using `flutter run` and you shall be seeing this error in the console:

> The plugin `cloud_functions` uses a deprecated version of the Android embedding.
To avoid unexpected runtime failures, or future build failures, try to see if this plugin supports the Android V2 embedding. Otherwise, consider removing it since a future release of Flutter will remove these deprecated APIs.
If you are plugin author, take a look at the docs for migrating the plugin to the V2 embedding: https://flutter.dev/go/android-plugin-migration.

### Plugin version:

```
cloud_functions: ^3.0.0
```

### Flutter version: 
```dart
Flutter 2.5.0-1.0.pre • channel dev • https://github.com/flutter/flutter.git
Framework • revision 184e5871be (10 days ago)
Tools • Dart 2.14.0 (build 2.14.0-350.0.dev)
```

**Comment by darshankawar** [comment]
Verified using latest plugin version and I see the same warning while running `flutter run`:

```
The plugin cloud_functions uses a deprecated version of the Android embedding.
To avoid unexpected runtime failures, or future build failures, try to see if this plugin supports the Android V2 embedding. Otherwise, consider removing it since a future release of Flutter will remove these deprecated APIs.
If you are plugin author, take a look at the docs for migrating the plugin to the V2 embedding: https://flutter.dev/go/android-plugin-migration.
```

**Comment by adfaous** [comment]
Same issue, any help?

**Comment by walsha2** [comment]
Can confirm, having the same exact issue as soon as I updated to Flutter SDK 2.5 on `stable` (released today).  

Since Flutter SDK 2.5 is now on `stable` can we make this issue a **priority to resolve**? A lot more users will run into this as soon as they run `flutter upgrade` on `stable`.

**Comment by Salakar** [comment]
Fixed in 3.0.3
